bundles/org.openhab.binding.vizio/src/main/java/org/openhab/binding/vizio/internal/handler/VizioHandler.java

```diff
@@ -0,0 +1,572 @@
+            Configuration configuration = this.getConfig();
+            configuration.put(PROPERTY_APP_LIST_JSON, appListJson);
+            this.updateConfiguration(configuration);
+        }
+
+        try {
+            VizioApps appsFromJson = gson.fromJson(appListJson, VizioApps.class);
+            if (appsFromJson != null && !appsFromJson.getApps().isEmpty()) {
+                userConfigApps = appsFromJson.getApps();
+
+                List<StateOption> appListOptions = new ArrayList<>();
+                userConfigApps.forEach(app -> {
+                    appListOptions.add(new StateOption(app.getName(), app.getName()));
+                });
+
+                stateDescriptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), ACTIVE_APP),
+                        appListOptions);
+            }
+        } catch (JsonSyntaxException e) {
+            logger.debug("Invalid App List Configuration in thing configuration. Exception: {}", e.getMessage(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                    "Invalid App List Configuration in thing configuration");
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        startVizioStateRefresh();
+        startPeriodicRefresh();
+    }
+
+    /**
+     * Start the job that queries the Vizio TV every 10 seconds to get its current status
+     */
+    private void startVizioStateRefresh() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            this.refreshJob = scheduler.scheduleWithFixedDelay(this::refreshVizioState, 5, 10, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Get current status from the Vizio TV and update the channels
+     */
+    private void refreshVizioState() {
+        synchronized (sequenceLock) {
+            try {
+                PowerMode polledPowerMode = communicator.getPowerMode();
+
+                if (debounce && !polledPowerMode.getItems().isEmpty()) {
+                    int powerMode = polledPowerMode.getItems().get(0).getValue();
+                    if (powerMode == 1) {
+                        powerOn = true;
+                        updateState(POWER, OnOffType.ON);
+                    } else if (powerMode == 0) {
+                        powerOn = false;
+                        updateState(POWER, OnOffType.OFF);
+                    } else {
+                        logger.debug("Unknown power mode {}, for response object: {}", powerMode, polledPowerMode);
+                    }
+                }
+                updateStatus(ThingStatus.ONLINE);
+            } catch (VizioException e) {
+                logger.debug("Unable to retrieve Vizio TV power mode info. Exception: {}", e.getMessage(), e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
Consider if the message above or even the exception message should be included in this status update to provide more immediate information to the user.